{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook will download all data. We assume the working directory is the root cibog dir!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from IPython.display import Image\n",
    "\n",
    "\n",
    "os.chdir(\"/mnt/w/game_cibog/\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/w/game_cibog\n"
     ]
    }
   ],
   "source": [
    "! pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Downloading and preparing data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Bulk Data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Downloading and processing bulk data from reads"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to be able to process the bulk data from reads we need several softwares and references.\n",
    "\n",
    "Make sure to install hisat2 and samtools (e.g. sudo apt-get install hisat2/samtools in Ubuntu).\n",
    "\n",
    "We also need featureCounts which is part of the subreads package: https://sourceforge.net/projects/subread/files/subread-2.0.3/ . Also make sure install a copy of that :)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-01-19 13:41:23--  https://jztkft.dl.sourceforge.net/project/subread/subread-2.0.3/subread-2.0.3-source.tar.gz\n",
      "Resolving jztkft.dl.sourceforge.net (jztkft.dl.sourceforge.net)... 45.67.159.245\n",
      "Connecting to jztkft.dl.sourceforge.net (jztkft.dl.sourceforge.net)|45.67.159.245|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 23304665 (22M) [application/x-gzip]\n",
      "Saving to: ‘subread-2.0.3-source.tar.gz’\n",
      "\n",
      "subread-2.0.3-sourc 100%[===================>]  22.22M  3.89MB/s    in 5.8s    \n",
      "\n",
      "2022-01-19 13:41:29 (3.81 MB/s) - ‘subread-2.0.3-source.tar.gz’ saved [23304665/23304665]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! cd tools && wget -O subread-2.0.3-source.tar.gz https://jztkft.dl.sourceforge.net/project/subread/subread-2.0.3/subread-2.0.3-source.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cd tools && tar xfz subread-2.0.3-source.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o core.o core.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o core-junction.o core-junction.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o core-indel.o core-indel.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o sambam-file.o sambam-file.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o sublog.o sublog.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o gene-algorithms.o gene-algorithms.c\n",
      "\u001b[01m\u001b[Kgene-algorithms.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kmiltime\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Kgene-algorithms.c:1585:3:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[Kftime\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
      " 1585 |   \u001b[01;35m\u001b[Kftime\u001b[m\u001b[K(&trp);\n",
      "      |   \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n",
      "In file included from \u001b[01m\u001b[Kgene-algorithms.c:33\u001b[m\u001b[K:\n",
      "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/sys/timeb.h:39:12:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
      "   39 | extern int \u001b[01;36m\u001b[Kftime\u001b[m\u001b[K (struct timeb *__timebuf)\n",
      "      |            \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o hashtable.o hashtable.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o input-files.o input-files.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o sorted-hashtable.o sorted-hashtable.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o gene-value-index.o gene-value-index.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o exon-algorithms.o exon-algorithms.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o HelperFunctions.o HelperFunctions.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o interval_merge.o interval_merge.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o long-hashtable.o long-hashtable.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o core-bigtable.o core-bigtable.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o seek-zlib.o seek-zlib.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb     -c -o input-blc.o input-blc.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o genRandomReads gen_rand_reads.c core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o detectionCall detection-calls.c  core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "echo \" \" > longread-one/make.version\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o repair read-repair.c  core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "rm -f longread-one/*.o\n",
      "cd longread-one && make\n",
      "make[1]: Entering directory '/mnt/w/game_cibog/tools/subread-2.0.3-source/src/longread-one'\n",
      "make[1]: Warning: File 'make.version' has modification time 0.6 s in the future\n",
      "gcc  -mtune=core2  -O3 -Wall  -DMAKE_FOR_EXON  -D MAKE_STANDALONE  -D_FILE_OFFSET_BITS=64 -DSUBREAD_VERSION=\\\"\"2.0.3\"\\\"    -fmessage-length=0  -ggdb -o LRMsorted-hashtable.o -c LRMsorted-hashtable.c\n",
      "gcc  -mtune=core2  -O3 -Wall  -DMAKE_FOR_EXON  -D MAKE_STANDALONE  -D_FILE_OFFSET_BITS=64 -DSUBREAD_VERSION=\\\"\"2.0.3\"\\\"    -fmessage-length=0  -ggdb -o LRMbase-index.o -c LRMbase-index.c\n",
      "gcc  -mtune=core2  -O3 -Wall  -DMAKE_FOR_EXON  -D MAKE_STANDALONE  -D_FILE_OFFSET_BITS=64 -DSUBREAD_VERSION=\\\"\"2.0.3\"\\\"    -fmessage-length=0  -ggdb -o LRMchro-event.o -c LRMchro-event.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o txUnique tx-unique.c  core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "gcc  -mtune=core2  -O3 -Wall  -DMAKE_FOR_EXON  -D MAKE_STANDALONE  -D_FILE_OFFSET_BITS=64 -DSUBREAD_VERSION=\\\"\"2.0.3\"\\\"    -fmessage-length=0  -ggdb -o LRMhelper.o -c LRMhelper.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o featureCounts readSummary.c core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "gcc  -mtune=core2  -O3 -Wall  -DMAKE_FOR_EXON  -D MAKE_STANDALONE  -D_FILE_OFFSET_BITS=64 -DSUBREAD_VERSION=\\\"\"2.0.3\"\\\"    -fmessage-length=0  -ggdb -o LRMseek-zlib.o -c LRMseek-zlib.c\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o  removeDup removeDupReads.c  core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "\u001b[01m\u001b[KreadSummary.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kfc_thread_merge_results\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[KreadSummary.c:6105:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      " 6105 |     \u001b[01;35m\u001b[Kfread(&section1_items,1, 8, input_fps[xk1])\u001b[m\u001b[K;\n",
      "      |     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KreadSummary.c:6108:6:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      " 6108 |      \u001b[01;35m\u001b[Kfread(&cellbcP0_geneno0B,1,8,input_fps[xk1])\u001b[m\u001b[K;\n",
      "      |      \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KreadSummary.c:6109:6:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      " 6109 |      \u001b[01;35m\u001b[Kfread(&umis,1,8,input_fps[xk1])\u001b[m\u001b[K;\n",
      "      |      \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KreadSummary.c:6130:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      " 6130 |     \u001b[01;35m\u001b[Kfread(last_rbin_buffer[xk1]+16, 1, 8*genes+ global_context -> scRNA_UMI_length + 4, input_fps[xk1])\u001b[m\u001b[K;\n",
      "      |     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KreadSummary.c:6132:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      " 6132 |     \u001b[01;35m\u001b[Kfread(last_rbin_buffer[xk1] + 16+ 8*genes+ global_context -> scRNA_UMI_length + 4, 1, binlen, input_fps[xk1])\u001b[m\u001b[K;\n",
      "      |     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KreadSummary.c:6190:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      " 6190 |     \u001b[01;35m\u001b[Kfread(last_rbin_buffer[selected_fp_no]+16, 1, 8*genes+ global_context -> scRNA_UMI_length + 4, input_fps[selected_fp_no])\u001b[m\u001b[K;\n",
      "      |     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KreadSummary.c:6193:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      " 6193 |     \u001b[01;35m\u001b[Kfread(last_rbin_buffer[selected_fp_no] + 16+ 8*genes+ global_context -> scRNA_UMI_length + 4, 1, binlen, input_fps[selected_fp_no])\u001b[m\u001b[K;\n",
      "      |     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KreadSummary.c:6240:4:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      " 6240 |    \u001b[01;35m\u001b[Kfread(&binlen, 1, 4, notmapped_fp)\u001b[m\u001b[K;\n",
      "      |    \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KreadSummary.c:6243:4:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      " 6243 |    \u001b[01;35m\u001b[Kfread(tofill -> inbin + tofill -> inbin_len, 1, binlen, notmapped_fp)\u001b[m\u001b[K;\n",
      "      |    \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "gcc  -mtune=core2  -O3 -Wall  -DMAKE_FOR_EXON  -D MAKE_STANDALONE  -D_FILE_OFFSET_BITS=64 -DSUBREAD_VERSION=\\\"\"2.0.3\"\\\"    -fmessage-length=0  -ggdb -o LRMfile-io.o -c LRMfile-io.c\n",
      "gcc  -mtune=core2  -O3 -Wall  -DMAKE_FOR_EXON  -D MAKE_STANDALONE  -D_FILE_OFFSET_BITS=64 -DSUBREAD_VERSION=\\\"\"2.0.3\"\\\"    -fmessage-length=0  -ggdb -o LRMhashtable.o -c LRMhashtable.c\n",
      "gcc  -mtune=core2  -O3 -Wall  -DMAKE_FOR_EXON  -D MAKE_STANDALONE  -D_FILE_OFFSET_BITS=64 -DSUBREAD_VERSION=\\\"\"2.0.3\"\\\"    -fmessage-length=0  -ggdb -o LRM longread-mapping.c   LRMsorted-hashtable.o LRMbase-index.o LRMchro-event.o LRMhelper.o LRMseek-zlib.o LRMfile-io.o LRMhashtable.o  -lpthread -lz -lm -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE \n",
      "\u001b[01m\u001b[Klongread-mapping.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KLRMmiltime\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Klongread-mapping.c:234:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[Kftime\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
      "  234 |  \u001b[01;35m\u001b[Kftime\u001b[m\u001b[K(&trp);\n",
      "      |  \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n",
      "In file included from \u001b[01m\u001b[Klongread-mapping.c:33\u001b[m\u001b[K:\n",
      "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/sys/timeb.h:39:12:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
      "   39 | extern int \u001b[01;36m\u001b[Kftime\u001b[m\u001b[K (struct timeb *__timebuf)\n",
      "      |            \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb   -o exactSNP SNPCalling.c  core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb   -o subread-buildindex  index-builder.c core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "make[1]: warning:  Clock skew detected.  Your build may be incomplete.\n",
      "make[1]: Leaving directory '/mnt/w/game_cibog/tools/subread-2.0.3-source/src/longread-one'\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o subindel SUBindel.c core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "\u001b[01m\u001b[KSUBindel.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kload_global_context_forindel\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[KSUBindel.c:66:24:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K.reads\u001b[m\u001b[K’ directive writing 6 bytes into a region of size between 1 and 1000 [\u001b[01;35m\u001b[K-Wformat-overflow=\u001b[m\u001b[K]\n",
      "   66 |  sprintf(tmp_fname, \"%s\u001b[01;35m\u001b[K.reads\u001b[m\u001b[K\", context->config.index_prefix);\n",
      "      |                        \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
      "In file included from \u001b[01m\u001b[K/usr/include/stdio.h:867\u001b[m\u001b[K,\n",
      "                 from \u001b[01m\u001b[KSUBindel.c:20\u001b[m\u001b[K:\n",
      "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/stdio2.h:36:10:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin___sprintf_chk\u001b[m\u001b[K’ output between 7 and 1006 bytes into a destination of size 1000\n",
      "   36 |   return \u001b[01;36m\u001b[K__builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,\u001b[m\u001b[K\n",
      "      |          \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "   37 | \u001b[01;36m\u001b[K      __bos (__s), __fmt, __va_arg_pack ())\u001b[m\u001b[K;\n",
      "      |       \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KSUBindel.c:77:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K.00.b.tab\u001b[m\u001b[K’ directive writing 9 bytes into a region of size between 1 and 1000 [\u001b[01;35m\u001b[K-Wformat-overflow=\u001b[m\u001b[K]\n",
      "   77 |   sprintf(tmp_fname, \"%s\u001b[01;35m\u001b[K.00.b.tab\u001b[m\u001b[K\", context->config.index_prefix);\n",
      "      |                         \u001b[01;35m\u001b[K^~~~~~~~~\u001b[m\u001b[K\n",
      "In file included from \u001b[01m\u001b[K/usr/include/stdio.h:867\u001b[m\u001b[K,\n",
      "                 from \u001b[01m\u001b[KSUBindel.c:20\u001b[m\u001b[K:\n",
      "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/stdio2.h:36:10:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin___sprintf_chk\u001b[m\u001b[K’ output between 10 and 1009 bytes into a destination of size 1000\n",
      "   36 |   return \u001b[01;36m\u001b[K__builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,\u001b[m\u001b[K\n",
      "      |          \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "   37 | \u001b[01;36m\u001b[K      __bos (__s), __fmt, __va_arg_pack ())\u001b[m\u001b[K;\n",
      "      |       \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KSUBindel.c:75:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K.00.c.tab\u001b[m\u001b[K’ directive writing 9 bytes into a region of size between 1 and 1000 [\u001b[01;35m\u001b[K-Wformat-overflow=\u001b[m\u001b[K]\n",
      "   75 |   sprintf(tmp_fname, \"%s\u001b[01;35m\u001b[K.00.c.tab\u001b[m\u001b[K\", context->config.index_prefix);\n",
      "      |                         \u001b[01;35m\u001b[K^~~~~~~~~\u001b[m\u001b[K\n",
      "In file included from \u001b[01m\u001b[K/usr/include/stdio.h:867\u001b[m\u001b[K,\n",
      "                 from \u001b[01m\u001b[KSUBindel.c:20\u001b[m\u001b[K:\n",
      "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/stdio2.h:36:10:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin___sprintf_chk\u001b[m\u001b[K’ output between 10 and 1009 bytes into a destination of size 1000\n",
      "   36 |   return \u001b[01;36m\u001b[K__builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,\u001b[m\u001b[K\n",
      "      |          \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "   37 | \u001b[01;36m\u001b[K      __bos (__s), __fmt, __va_arg_pack ())\u001b[m\u001b[K;\n",
      "      |       \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KSUBindel.c:87:26:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K%02d\u001b[m\u001b[K’ directive writing between 2 and 11 bytes into a region of size between 0 and 999 [\u001b[01;35m\u001b[K-Wformat-overflow=\u001b[m\u001b[K]\n",
      "   87 |   sprintf(tmp_fname, \"%s.\u001b[01;35m\u001b[K%02d\u001b[m\u001b[K.%c.tab\", context->config.index_prefix, context->index_block_number, context->config.space_type == GENE_SPACE_COLOR?'c':'b');\n",
      "      |                          \u001b[01;35m\u001b[K^~~~\u001b[m\u001b[K\n",
      "In file included from \u001b[01m\u001b[K/usr/include/stdio.h:867\u001b[m\u001b[K,\n",
      "                 from \u001b[01m\u001b[KSUBindel.c:20\u001b[m\u001b[K:\n",
      "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/stdio2.h:36:10:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin___sprintf_chk\u001b[m\u001b[K’ output between 10 and 1018 bytes into a destination of size 1000\n",
      "   36 |   return \u001b[01;36m\u001b[K__builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,\u001b[m\u001b[K\n",
      "      |          \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "   37 | \u001b[01;36m\u001b[K      __bos (__s), __fmt, __va_arg_pack ())\u001b[m\u001b[K;\n",
      "      |       \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KSUBindel.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kload_indexes_forindel\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[KSUBindel.c:110:26:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K%02d\u001b[m\u001b[K’ directive writing between 2 and 10 bytes into a region of size between 0 and 999 [\u001b[01;35m\u001b[K-Wformat-overflow=\u001b[m\u001b[K]\n",
      "  110 |   sprintf(tmp_fname, \"%s.\u001b[01;35m\u001b[K%02d\u001b[m\u001b[K.%c.array\", global_context->config.index_prefix, block_no,  global_context->config.space_type == GENE_SPACE_COLOR?'c':'b');\n",
      "      |                          \u001b[01;35m\u001b[K^~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[KSUBindel.c:110:22:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdirective argument in the range [0, 2147483647]\n",
      "  110 |   sprintf(tmp_fname, \u001b[01;36m\u001b[K\"%s.%02d.%c.array\"\u001b[m\u001b[K, global_context->config.index_prefix, block_no,  global_context->config.space_type == GENE_SPACE_COLOR?'c':'b');\n",
      "      |                      \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "In file included from \u001b[01m\u001b[K/usr/include/stdio.h:867\u001b[m\u001b[K,\n",
      "                 from \u001b[01m\u001b[KSUBindel.c:20\u001b[m\u001b[K:\n",
      "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/stdio2.h:36:10:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin___sprintf_chk\u001b[m\u001b[K’ output between 12 and 1019 bytes into a destination of size 1000\n",
      "   36 |   return \u001b[01;36m\u001b[K__builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,\u001b[m\u001b[K\n",
      "      |          \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "   37 | \u001b[01;36m\u001b[K      __bos (__s), __fmt, __va_arg_pack ())\u001b[m\u001b[K;\n",
      "      |       \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o subread-align core-interface-aligner.c  core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm \n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o subjunc core-interface-subjunc.c  core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm \n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o qualityScores qualityScores.c  core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o   -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o subread-fullscan fullscan.c  core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o   -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o propmapped propmapped.c core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "gcc  -mtune=core2  -O3 -DMAKE_FOR_EXON  -D MAKE_STANDALONE -D SUBREAD_VERSION=\\\"\"2.0.3\"\\\"  -D_FILE_OFFSET_BITS=64    -fmessage-length=0  -ggdb  -o flattenGTF flattenAnnotations.c  core.o core-junction.o core-indel.o sambam-file.o sublog.o gene-algorithms.o hashtable.o input-files.o sorted-hashtable.o gene-value-index.o exon-algorithms.o HelperFunctions.o interval_merge.o long-hashtable.o core-bigtable.o seek-zlib.o input-blc.o  -pthread -lz  -O3 -DMAKE_FOR_EXON -D MAKE_STANDALONE -lm\n",
      "mkdir -p ../bin/utilities\n",
      "mv longread-one/LRM longread-one/sublong\n",
      "mv longread-one/sublong subread-align subjunc featureCounts  subindel exactSNP subread-buildindex ../bin/\n",
      "mv detectionCall genRandomReads repair propmapped qualityScores removeDup subread-fullscan txUnique flattenGTF ../bin/utilities\n",
      "\n",
      "###########################################################\n",
      "#                                                         #\n",
      "#  Installation successfully completed.                   #\n",
      "#                                                         #\n",
      "#  Generated executables were copied to directory ../bin/ #\n",
      "#                                                         #\n",
      "###########################################################\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! cd tools/subread-2.0.3-source/src && make -f Makefile.Linux -j 4"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can now check that featureCounts was built and is executable:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "featureCounts v2.0.3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! tools/subread-2.0.3-source/bin/featureCounts -v"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now that everything is installed, let's download and rename the files! At the end we will download all required references (genome + genome annotation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 17872256 spots for SRR9048093\n",
      "Written 17872256 spots for SRR9048093\n",
      "Read 40740514 spots for SRR9048094\n",
      "Written 40740514 spots for SRR9048094\n",
      "Read 43554379 spots for SRR9048103\n",
      "Written 43554379 spots for SRR9048103\n",
      "Read 28277603 spots for SRR9048104\n",
      "Written 28277603 spots for SRR9048104\n"
     ]
    }
   ],
   "source": [
    "#24h R1\n",
    "! cd reads/bulk/ && fastq-dump --gzip SRR9048093\n",
    "#24h R1\n",
    "! cd reads/bulk/ && fastq-dump --gzip SRR9048094\n",
    "#JQ1 24h R1\n",
    "! cd reads/bulk/ && fastq-dump --gzip SRR9048103\n",
    "#JQ1 24h R1\n",
    "! cd reads/bulk/ && fastq-dump --gzip SRR9048104\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mv reads/bulk/SRR9048093.fastq.gz reads/bulk/SUM159_24h_R1.fastq.gz\n",
    "! mv reads/bulk/SRR9048094.fastq.gz reads/bulk/SUM159_24h_R2.fastq.gz\n",
    "! mv reads/bulk/SRR9048103.fastq.gz reads/bulk/SUM159R_24h_R1.fastq.gz\n",
    "! mv reads/bulk/SRR9048104.fastq.gz reads/bulk/SUM159R_24h_R2.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-01-19 12:36:56--  https://genome-idx.s3.amazonaws.com/hisat/grch38_genome.tar.gz\n",
      "Resolving genome-idx.s3.amazonaws.com (genome-idx.s3.amazonaws.com)... 52.217.18.212\n",
      "Connecting to genome-idx.s3.amazonaws.com (genome-idx.s3.amazonaws.com)|52.217.18.212|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 4210306865 (3.9G) [binary/octet-stream]\n",
      "Saving to: ‘hsapien_hisat2.tar.gz’\n",
      "\n",
      "hsapien_hisat2.tar. 100%[===================>]   3.92G  4.36MB/s    in 16m 19s \n",
      "\n",
      "2022-01-19 12:53:16 (4.10 MB/s) - ‘hsapien_hisat2.tar.gz’ saved [4210306865/4210306865]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! cd references && wget -O hsapiens_hisat2.tar.gz https://genome-idx.s3.amazonaws.com/hisat/grch38_genome.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cd references && tar xfz hsapien_hisat2.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-01-19 14:24:48--  http://ftp.ensembl.org/pub/release-105/gtf/homo_sapiens/Homo_sapiens.GRCh38.105.gtf.gz\n",
      "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.197.76\n",
      "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.197.76|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 50829040 (48M) [application/x-gzip]\n",
      "Saving to: ‘ensembl.105.annotation.gtf.gz’\n",
      "\n",
      "ensembl.105.annotat 100%[===================>]  48.47M  1.90MB/s    in 26s     \n",
      "\n",
      "2022-01-19 14:25:14 (1.90 MB/s) - ‘ensembl.105.annotation.gtf.gz’ saved [50829040/50829040]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! cd references && wget -O ensembl.105.annotation.gtf.gz http://ftp.ensembl.org/pub/release-105/gtf/homo_sapiens/Homo_sapiens.GRCh38.105.gtf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cd references && gunzip ensembl.105.annotation.gtf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download preprocessed gene expression files from GEO\n",
    "\n",
    "First the 24h samples:\n",
    "\n",
    "<img src=\"../images/bulk_sample_gse.png\" />"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-01-18 14:08:01--  https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763467&format=file&file=GSM3763467%5FSUM159%5FDMSO%5F24h%5FR1%2Ecounts%2Etxt%2Egz\n",
      "Resolving www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)... 130.14.29.110, 2607:f220:41e:4290::110\n",
      "Connecting to www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)|130.14.29.110|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 96895 (95K) [application/octet-stream]\n",
      "Saving to: ‘GSM3763467_SUM159_DMSO_24h_R1.counts.txt.gz’\n",
      "\n",
      "GSM3763467_SUM159_D 100%[===================>]  94.62K   420KB/s    in 0.2s    \n",
      "\n",
      "2022-01-18 14:08:02 (420 KB/s) - ‘GSM3763467_SUM159_DMSO_24h_R1.counts.txt.gz’ saved [96895/96895]\n",
      "\n",
      "--2022-01-18 14:08:02--  https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763468&format=file&file=GSM3763468%5FSUM159%5FDMSO%5F24h%5FR2%2Ecounts%2Etxt%2Egz\n",
      "Resolving www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)... 130.14.29.110, 2607:f220:41e:4290::110\n",
      "Connecting to www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)|130.14.29.110|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 101271 (99K) [application/octet-stream]\n",
      "Saving to: ‘GSM3763468_SUM159_DMSO_24h_R2.counts.txt.gz’\n",
      "\n",
      "GSM3763468_SUM159_D 100%[===================>]  98.90K   313KB/s    in 0.3s    \n",
      "\n",
      "2022-01-18 14:08:02 (313 KB/s) - ‘GSM3763468_SUM159_DMSO_24h_R2.counts.txt.gz’ saved [101271/101271]\n",
      "\n",
      "--2022-01-18 14:08:03--  https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763477&format=file&file=GSM3763477%5FSUM159%5FJQ1R%5FDMSO%5F24h%5FR1%2Ecounts%2Etxt%2Egz\n",
      "Resolving www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)... 130.14.29.110, 2607:f220:41e:4290::110\n",
      "Connecting to www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)|130.14.29.110|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 100161 (98K) [application/octet-stream]\n",
      "Saving to: ‘GSM3763477_SUM159_JQ1R_DMSO_24h_R1.counts.txt.gz’\n",
      "\n",
      "GSM3763477_SUM159_J 100%[===================>]  97.81K   312KB/s    in 0.3s    \n",
      "\n",
      "2022-01-18 14:08:03 (312 KB/s) - ‘GSM3763477_SUM159_JQ1R_DMSO_24h_R1.counts.txt.gz’ saved [100161/100161]\n",
      "\n",
      "--2022-01-18 14:08:03--  https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763478&format=file&file=GSM3763478%5FSUM159%5FJQ1R%5FDMSO%5F24h%5FR2%2Ecounts%2Etxt%2Egz\n",
      "Resolving www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)... 130.14.29.110, 2607:f220:41e:4290::110\n",
      "Connecting to www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)|130.14.29.110|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 97689 (95K) [application/octet-stream]\n",
      "Saving to: ‘GSM3763478_SUM159_JQ1R_DMSO_24h_R2.counts.txt.gz’\n",
      "\n",
      "GSM3763478_SUM159_J 100%[===================>]  95.40K   410KB/s    in 0.2s    \n",
      "\n",
      "2022-01-18 14:08:04 (410 KB/s) - ‘GSM3763478_SUM159_JQ1R_DMSO_24h_R2.counts.txt.gz’ saved [97689/97689]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! cd bulk && wget -O GSM3763467_SUM159_DMSO_24h_R1.counts.txt.gz \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763467&format=file&file=GSM3763467%5FSUM159%5FDMSO%5F24h%5FR1%2Ecounts%2Etxt%2Egz\"\n",
    "! cd bulk && wget -O GSM3763468_SUM159_DMSO_24h_R2.counts.txt.gz \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763468&format=file&file=GSM3763468%5FSUM159%5FDMSO%5F24h%5FR2%2Ecounts%2Etxt%2Egz\"\n",
    "\n",
    "! cd bulk && wget -O GSM3763477_SUM159_JQ1R_DMSO_24h_R1.counts.txt.gz \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763477&format=file&file=GSM3763477%5FSUM159%5FJQ1R%5FDMSO%5F24h%5FR1%2Ecounts%2Etxt%2Egz\"\n",
    "! cd bulk && wget -O GSM3763478_SUM159_JQ1R_DMSO_24h_R2.counts.txt.gz \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763478&format=file&file=GSM3763478%5FSUM159%5FJQ1R%5FDMSO%5F24h%5FR2%2Ecounts%2Etxt%2Egz\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then the 3h samples:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-01-18 14:08:19--  https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763469&format=file&file=GSM3763469%5FSUM159%5FDMSO%5F3h%5FR1%2Ecounts%2Etxt%2Egz\n",
      "Resolving www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)... 130.14.29.110, 2607:f220:41e:4290::110\n",
      "Connecting to www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)|130.14.29.110|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 101284 (99K) [application/octet-stream]\n",
      "Saving to: ‘GSM3763469_SUM159_DMSO_3h_R1.counts.txt.gz’\n",
      "\n",
      "GSM3763469_SUM159_D 100%[===================>]  98.91K   312KB/s    in 0.3s    \n",
      "\n",
      "2022-01-18 14:08:20 (312 KB/s) - ‘GSM3763469_SUM159_DMSO_3h_R1.counts.txt.gz’ saved [101284/101284]\n",
      "\n",
      "--2022-01-18 14:08:20--  https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763470&format=file&file=GSM3763470%5FSUM159%5FDMSO%5F3h%5FR2%2Ecounts%2Etxt%2Egz\n",
      "Resolving www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)... 130.14.29.110, 2607:f220:41e:4290::110\n",
      "Connecting to www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)|130.14.29.110|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 97700 (95K) [application/octet-stream]\n",
      "Saving to: ‘GSM3763470_SUM159_DMSO_3h_R2.counts.txt.gz’\n",
      "\n",
      "GSM3763470_SUM159_D 100%[===================>]  95.41K   432KB/s    in 0.2s    \n",
      "\n",
      "2022-01-18 14:08:21 (432 KB/s) - ‘GSM3763470_SUM159_DMSO_3h_R2.counts.txt.gz’ saved [97700/97700]\n",
      "\n",
      "--2022-01-18 14:08:21--  https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763479&format=file&file=GSM3763479%5FSUM159%5FJQ1R%5FDMSO%5F3h%5FR1%2Ecounts%2Etxt%2Egz\n",
      "Resolving www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)... 130.14.29.110, 2607:f220:41e:4290::110\n",
      "Connecting to www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)|130.14.29.110|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 99026 (97K) [application/octet-stream]\n",
      "Saving to: ‘GSM3763479_SUM159_JQ1R_DMSO_3h_R1.counts.txt.gz’\n",
      "\n",
      "GSM3763479_SUM159_J 100%[===================>]  96.71K   304KB/s    in 0.3s    \n",
      "\n",
      "2022-01-18 14:08:22 (304 KB/s) - ‘GSM3763479_SUM159_JQ1R_DMSO_3h_R1.counts.txt.gz’ saved [99026/99026]\n",
      "\n",
      "--2022-01-18 14:08:22--  https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763480&format=file&file=GSM3763480%5FSUM159%5FJQ1R%5FDMSO%5F3h%5FR2%2Ecounts%2Etxt%2Egz\n",
      "Resolving www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)... 130.14.29.110, 2607:f220:41e:4290::110\n",
      "Connecting to www.ncbi.nlm.nih.gov (www.ncbi.nlm.nih.gov)|130.14.29.110|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 99791 (97K) [application/octet-stream]\n",
      "Saving to: ‘GSM3763480_SUM159_JQ1R_DMSO_3h_R2.counts.txt.gz’\n",
      "\n",
      "GSM3763480_SUM159_J 100%[===================>]  97.45K   313KB/s    in 0.3s    \n",
      "\n",
      "2022-01-18 14:08:23 (313 KB/s) - ‘GSM3763480_SUM159_JQ1R_DMSO_3h_R2.counts.txt.gz’ saved [99791/99791]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! cd bulk && wget -O GSM3763469_SUM159_DMSO_3h_R1.counts.txt.gz \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763469&format=file&file=GSM3763469%5FSUM159%5FDMSO%5F3h%5FR1%2Ecounts%2Etxt%2Egz\"\n",
    "! cd bulk && wget -O GSM3763470_SUM159_DMSO_3h_R2.counts.txt.gz \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763470&format=file&file=GSM3763470%5FSUM159%5FDMSO%5F3h%5FR2%2Ecounts%2Etxt%2Egz\"\n",
    "\n",
    "! cd bulk && wget -O GSM3763479_SUM159_JQ1R_DMSO_3h_R1.counts.txt.gz \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763479&format=file&file=GSM3763479%5FSUM159%5FJQ1R%5FDMSO%5F3h%5FR1%2Ecounts%2Etxt%2Egz\"\n",
    "! cd bulk && wget -O GSM3763480_SUM159_JQ1R_DMSO_3h_R2.counts.txt.gz \"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM3763480&format=file&file=GSM3763480%5FSUM159%5FJQ1R%5FDMSO%5F3h%5FR2%2Ecounts%2Etxt%2Egz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24h_sum159_vs_sum159_jq1r.tsv\n",
      "3h_sum159_vs_sum159_jq1r.tsv\n",
      "GSM3763467_SUM159_DMSO_24h_R1.counts.txt.gz\n",
      "GSM3763468_SUM159_DMSO_24h_R2.counts.txt.gz\n",
      "GSM3763469_SUM159_DMSO_3h_R1.counts.txt.gz\n",
      "GSM3763470_SUM159_DMSO_3h_R2.counts.txt.gz\n",
      "GSM3763477_SUM159_JQ1R_DMSO_24h_R1.counts.txt.gz\n",
      "GSM3763478_SUM159_JQ1R_DMSO_24h_R2.counts.txt.gz\n",
      "GSM3763479_SUM159_JQ1R_DMSO_3h_R1.counts.txt.gz\n",
      "GSM3763480_SUM159_JQ1R_DMSO_3h_R2.counts.txt.gz\n",
      "normed_expr.24h_sum159_vs_sum159_jq1r.tsv\n",
      "normed_expr.3h_sum159_vs_sum159_jq1r.tsv\n",
      "sig.24h_sum159_vs_sum159_jq1r.tsv\n",
      "sig.3h_sum159_vs_sum159_jq1r.tsv\n",
      "sig.down.24h_sum159_vs_sum159_jq1r.tsv\n",
      "sig.down.3h_sum159_vs_sum159_jq1r.tsv\n",
      "sig.sum159_vs_sum159_jq1r.tsv\n",
      "sig.up.24h_sum159_vs_sum159_jq1r.tsv\n",
      "sig.up.3h_sum159_vs_sum159_jq1r.tsv\n",
      "top.all.24h_sum159_vs_sum159_jq1r.tsv\n",
      "top.all.3h_sum159_vs_sum159_jq1r.tsv\n",
      "top.down.24h_sum159_vs_sum159_jq1r.tsv\n",
      "top.down.3h_sum159_vs_sum159_jq1r.tsv\n",
      "top.up.24h_sum159_vs_sum159_jq1r.tsv\n",
      "top.up.3h_sum159_vs_sum159_jq1r.tsv\n"
     ]
    }
   ],
   "source": [
    "! ls bulk"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download scRNA-seq data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next we can download the scRNA-seq data."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For this we first need to install sratoolkit for downloading SRA-hosted files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! wget --output-document tools/sratoolkit.tar.gz http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz\n",
    "\n",
    "! cd tools && tar -vxzf sratoolkit.tar.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Briefly run vdb-config and exit without changes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! vdb-config --interactive"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can already download cellranger at this time. Cellranger is required for processing 10X Genomics datasets. In this case, we also need the human reference genome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-01-11 18:27:18--  https://cf.10xgenomics.com/releases/cell-exp/cellranger-6.1.2.tar.gz?Expires=1641958065&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9jZi4xMHhnZW5vbWljcy5jb20vcmVsZWFzZXMvY2VsbC1leHAvY2VsbHJhbmdlci02LjEuMi50YXIuZ3oiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE2NDE5NTgwNjV9fX1dfQ__&Signature=g2re3W~2ybB0B3sFtdXmM~W2fWTUF4AMKF~46oEqQ0lV9W4IBfJ-7kpnSgAHj7KbIsB-Pu7l2l3gX8i8evLnnmru3x8i6sp72K5eRYgpY7JxypcTuWi9sbF1p8ezHoL5C9dsR0QI3Gx9ZDbu2VA5wrGV7wRi0ehnbXTsUAjOnG87lKAq7khwC~HvcOux9Og8B0WzbSb2Hb1klEoAj5n5Xd~-ta8EoqjeqVdgue0-DVXD36kvU5OkYWOTehNZtTkXxBdDwgbjbKw-8XgZYnie1p8lfDWTZRU16o81-lovv0yq3XFkvh89Kuj84GjwxU6s-~d9Xq-BE8g28DG8vLlGxg__&Key-Pair-Id=APKAI7S6A5RYOXBWRPDA\n",
      "Resolving cf.10xgenomics.com (cf.10xgenomics.com)... 104.18.0.173, 104.18.1.173, 2606:4700::6812:ad, ...\n",
      "Connecting to cf.10xgenomics.com (cf.10xgenomics.com)|104.18.0.173|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 804457389 (767M) [application/x-tar]\n",
      "Saving to: ‘cellranger-6.1.2.tar.gz’\n",
      "\n",
      "cellranger-6.1.2.ta   2%[                    ]  20.57M   111KB/s    eta 36m 38s^C\n"
     ]
    }
   ],
   "source": [
    "! cd tools && wget -O cellranger-6.1.2.tar.gz \"https://cf.10xgenomics.com/releases/cell-exp/cellranger-6.1.2.tar.gz?Expires=1641958065&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9jZi4xMHhnZW5vbWljcy5jb20vcmVsZWFzZXMvY2VsbC1leHAvY2VsbHJhbmdlci02LjEuMi50YXIuZ3oiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE2NDE5NTgwNjV9fX1dfQ__&Signature=g2re3W~2ybB0B3sFtdXmM~W2fWTUF4AMKF~46oEqQ0lV9W4IBfJ-7kpnSgAHj7KbIsB-Pu7l2l3gX8i8evLnnmru3x8i6sp72K5eRYgpY7JxypcTuWi9sbF1p8ezHoL5C9dsR0QI3Gx9ZDbu2VA5wrGV7wRi0ehnbXTsUAjOnG87lKAq7khwC~HvcOux9Og8B0WzbSb2Hb1klEoAj5n5Xd~-ta8EoqjeqVdgue0-DVXD36kvU5OkYWOTehNZtTkXxBdDwgbjbKw-8XgZYnie1p8lfDWTZRU16o81-lovv0yq3XFkvh89Kuj84GjwxU6s-~d9Xq-BE8g28DG8vLlGxg__&Key-Pair-Id=APKAI7S6A5RYOXBWRPDA\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cd tools && tar xfz cellranger-6.1.2.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cd references && wget https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-GRCh38-2020-A.tar.gz && tar xfz refdata-gex-GRCh38-2020-A.tar.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We should briefly check whether cellranger can be executed:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cellranger sitecheck (cellranger-6.1.2)\n",
      "Copyright (c) 2021 10x Genomics, Inc.  All rights reserved.\n",
      "-------------------------------------------------------------------------------\n",
      "Wed Jan 12 09:48:36 CET 2022\n",
      "\n",
      "=====================================================================\n",
      "System Info\n",
      "uname -a\n",
      "---------------------------------------------------------------------\n",
      "Linux spectre3 4.19.104-microsoft-standard #1 SMP Wed Feb 19 06:37:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "Linux Distro\n",
      "cat /etc/*-release | sort -u\n",
      "---------------------------------------------------------------------\n",
      "BUG_REPORT_URL=\"https://bugs.launchpad.net/ubuntu/\"\n",
      "DISTRIB_CODENAME=focal\n",
      "DISTRIB_DESCRIPTION=\"Ubuntu 20.04.2 LTS\"\n",
      "DISTRIB_ID=Ubuntu\n",
      "DISTRIB_RELEASE=20.04\n",
      "HOME_URL=\"https://www.ubuntu.com/\"\n",
      "ID=ubuntu\n",
      "ID_LIKE=debian\n",
      "NAME=\"Ubuntu\"\n",
      "PRETTY_NAME=\"Ubuntu 20.04.2 LTS\"\n",
      "PRIVACY_POLICY_URL=\"https://www.ubuntu.com/legal/terms-and-policies/privacy-policy\"\n",
      "SUPPORT_URL=\"https://help.ubuntu.com/\"\n",
      "UBUNTU_CODENAME=focal\n",
      "VERSION=\"20.04.2 LTS (Focal Fossa)\"\n",
      "VERSION_CODENAME=focal\n",
      "VERSION_ID=\"20.04\"\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "Kernel Build\n",
      "cat /proc/version\n",
      "---------------------------------------------------------------------\n",
      "Linux version 4.19.104-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Wed Feb 19 06:37:35 UTC 2020\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "glibc version\n",
      "ldd --version | head -n 1\n",
      "---------------------------------------------------------------------\n",
      "ldd (Ubuntu GLIBC 2.31-0ubuntu9.2) 2.31\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "CPU Model\n",
      "grep -m 1 'model name' /proc/cpuinfo | cut -d ':' -f 2 | sed 's/^[ \\t]*//'\n",
      "---------------------------------------------------------------------\n",
      "AMD Ryzen 7 3700X 8-Core Processor\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "CPU Sockets\n",
      "grep 'physical id' /proc/cpuinfo | sort -u | wc -l\n",
      "---------------------------------------------------------------------\n",
      "1\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "CPU Cores\n",
      "grep -c processor /proc/cpuinfo\n",
      "---------------------------------------------------------------------\n",
      "16\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "CPU Support\n",
      "grep -m 1 'flags' /proc/cpuinfo | cut -d ':' -f 2 | sed 's/^\\s*//'\n",
      "---------------------------------------------------------------------\n",
      "fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm rep_good nopl cpuid extd_apicid pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext ssbd ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves clzero xsaveerptr virt_ssbd arat umip rdpid\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "Memory Total\n",
      "grep MemTotal /proc/meminfo | cut -d ':' -f 2 | sed 's/^[ \\t]*//'\n",
      "---------------------------------------------------------------------\n",
      "52582416 kB\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "Disk Space\n",
      "df -Ph | awk '{print $2, $3, $4}'\n",
      "---------------------------------------------------------------------\n",
      "Size Used Avail\n",
      "251G 95G 144G\n",
      "26G 365M 25G\n",
      "251G 7.0G 232G\n",
      "26G 12K 26G\n",
      "251G 123M 239G\n",
      "370M 370M 0\n",
      "238G 190G 49G\n",
      "26G 0 26G\n",
      "26G 4.0K 26G\n",
      "26G 0 26G\n",
      "26G 0 26G\n",
      "26G 0 26G\n",
      "26G 0 26G\n",
      "238G 190G 49G\n",
      "238G 190G 49G\n",
      "238G 190G 49G\n",
      "932G 852G 80G\n",
      "1.9T 945G 919G\n",
      "932G 849G 83G\n",
      "1.9T 1.1T 759G\n",
      "1.9T 33G 1.8T\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "Filesystem Options\n",
      "mount | cut -d ' ' -f 5,6\n",
      "---------------------------------------------------------------------\n",
      "ext4 (rw,relatime,discard,errors=remount-ro,data=ordered)\n",
      "tmpfs (rw,relatime)\n",
      "ext4 (rw,relatime,discard,errors=remount-ro,data=ordered)\n",
      "9p (rw,noatime,dirsync,aname=drvfs;path=T:\\;uid=1000;gid=1000;metadata;umask=22;fmask=11;symlinkroot=/mnt/,mmap,access=client,msize=65536,trans=fd,rfd=8,wfd=8)\n",
      "tmpfs (rw,nosuid,noexec,noatime,mode=755)\n",
      "tmpfs (rw,nosuid,noexec,noatime,mode=755)\n",
      "ext4 (rw,relatime,discard,errors=remount-ro,data=ordered)\n",
      "iso9660 (ro,relatime,nojoliet,check=s,map=n,blocksize=2048)\n",
      "tmpfs (rw,relatime)\n",
      "tmpfs (rw,relatime)\n",
      "tmpfs (rw,relatime)\n",
      "tmpfs (rw,relatime)\n",
      "tmpfs (rw,relatime)\n",
      "tmpfs (rw,relatime)\n",
      "tmpfs (rw,relatime)\n",
      "overlay (rw,relatime,lowerdir=/containers/services/docker/lower,upperdir=/containers/services/docker/tmp/upper,workdir=/containers/services/docker/tmp/work)\n",
      "overlay (rw,relatime,lowerdir=/containers/services/docker/lower,upperdir=/containers/services/docker/tmp/upper,workdir=/containers/services/docker/tmp/work)\n",
      "overlay (rw,relatime,lowerdir=/containers/services/docker/lower,upperdir=/containers/services/docker/tmp/upper,workdir=/containers/services/docker/tmp/work)\n",
      "overlay (rw,relatime,lowerdir=/containers/services/docker/lower,upperdir=/containers/services/docker/tmp/upper,workdir=/containers/services/docker/tmp/work)\n",
      "overlay (rw,relatime,lowerdir=/containers/services/docker/lower,upperdir=/containers/services/docker/tmp/upper,workdir=/containers/services/docker/tmp/work)\n",
      "overlay (rw,relatime,lowerdir=/containers/services/docker/lower,upperdir=/containers/services/docker/tmp/upper,workdir=/containers/services/docker/tmp/work)\n",
      "overlay (rw,relatime,lowerdir=/containers/services/docker/lower,upperdir=/containers/services/docker/tmp/upper,workdir=/containers/services/docker/tmp/work)\n",
      "9p (ro,relatime,dirsync,aname=tools;fmask=022,loose,access=client,trans=fd,rfd=6,wfd=6)\n",
      "devtmpfs (rw,nosuid,relatime,size=26288800k,nr_inodes=6572200,mode=755)\n",
      "sysfs (rw,nosuid,nodev,noexec,noatime)\n",
      "proc (rw,nosuid,nodev,noexec,noatime)\n",
      "devpts (rw,nosuid,noexec,noatime,gid=5,mode=620,ptmxmode=000)\n",
      "tmpfs (rw,nosuid,noexec,noatime,mode=755)\n",
      "tmpfs (rw,nosuid,nodev,noexec,noatime)\n",
      "tmpfs (rw,nosuid,nodev,noatime)\n",
      "tmpfs (rw,nosuid,nodev,noexec,noatime,mode=755)\n",
      "binfmt_misc (rw,relatime)\n",
      "tmpfs (rw,nosuid,nodev,noexec,relatime,mode=755)\n",
      "9p (ro,nosuid,nodev,noatime,dirsync,aname=drivers;fmask=222;dmask=222,mmap,access=client,msize=65536,trans=fd,rfd=4,wfd=4)\n",
      "9p (ro,nosuid,nodev,noatime,dirsync,aname=lib;fmask=222;dmask=222,mmap,access=client,msize=65536,trans=fd,rfd=4,wfd=4)\n",
      "cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,cpu)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,cpuacct)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,blkio)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,memory)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,devices)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,freezer)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,net_cls)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,net_prio)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,hugetlb)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,pids)\n",
      "cgroup (rw,nosuid,nodev,noexec,relatime,rdma)\n",
      "9p (rw,noatime,dirsync,aname=drvfs;path=C:\\;uid=1000;gid=1000;metadata;umask=22;fmask=11;symlinkroot=/mnt/,mmap,access=client,msize=65536,trans=fd,rfd=8,wfd=8)\n",
      "9p (rw,noatime,dirsync,aname=drvfs;path=F:\\;uid=1000;gid=1000;metadata;umask=22;fmask=11;symlinkroot=/mnt/,mmap,access=client,msize=65536,trans=fd,rfd=8,wfd=8)\n",
      "9p (rw,noatime,dirsync,aname=drvfs;path=G:\\;uid=1000;gid=1000;metadata;umask=22;fmask=11;symlinkroot=/mnt/,mmap,access=client,msize=65536,trans=fd,rfd=8,wfd=8)\n",
      "9p (rw,noatime,dirsync,aname=drvfs;path=S:\\;uid=1000;gid=1000;metadata;umask=22;fmask=11;symlinkroot=/mnt/,mmap,access=client,msize=65536,trans=fd,rfd=8,wfd=8)\n",
      "9p (rw,noatime,dirsync,aname=drvfs;path=T:\\;uid=1000;gid=1000;metadata;umask=22;fmask=11;symlinkroot=/mnt/,mmap,access=client,msize=65536,trans=fd,rfd=8,wfd=8)\n",
      "9p (rw,noatime,dirsync,aname=drvfs;path=W:\\;uid=1000;gid=1000;metadata;umask=22;fmask=11;symlinkroot=/mnt/,mmap,access=client,msize=65536,trans=fd,rfd=8,wfd=8)\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "User Limits\n",
      "bash -c 'ulimit -a'\n",
      "---------------------------------------------------------------------\n",
      "core file size          (blocks, -c) 0\n",
      "data seg size           (kbytes, -d) unlimited\n",
      "scheduling priority             (-e) 0\n",
      "file size               (blocks, -f) unlimited\n",
      "pending signals                 (-i) 205381\n",
      "max locked memory       (kbytes, -l) 64\n",
      "max memory size         (kbytes, -m) unlimited\n",
      "open files                      (-n) 4096\n",
      "pipe size            (512 bytes, -p) 8\n",
      "POSIX message queues     (bytes, -q) 819200\n",
      "real-time priority              (-r) 0\n",
      "stack size              (kbytes, -s) 8192\n",
      "cpu time               (seconds, -t) unlimited\n",
      "max user processes              (-u) 205381\n",
      "virtual memory          (kbytes, -v) unlimited\n",
      "file locks                      (-x) unlimited\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "User Limits (hard)\n",
      "bash -c 'ulimit -aH'\n",
      "---------------------------------------------------------------------\n",
      "core file size          (blocks, -c) unlimited\n",
      "data seg size           (kbytes, -d) unlimited\n",
      "scheduling priority             (-e) 0\n",
      "file size               (blocks, -f) unlimited\n",
      "pending signals                 (-i) 205381\n",
      "max locked memory       (kbytes, -l) 64\n",
      "max memory size         (kbytes, -m) unlimited\n",
      "open files                      (-n) 4096\n",
      "pipe size            (512 bytes, -p) 8\n",
      "POSIX message queues     (bytes, -q) 819200\n",
      "real-time priority              (-r) 0\n",
      "stack size              (kbytes, -s) unlimited\n",
      "cpu time               (seconds, -t) unlimited\n",
      "max user processes              (-u) 205381\n",
      "virtual memory          (kbytes, -v) unlimited\n",
      "file locks                      (-x) unlimited\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "Global File Limit\n",
      "cat /proc/sys/fs/file-{max,nr}\n",
      "---------------------------------------------------------------------\n",
      "5256606\n",
      "1856\t0\t5256606\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "Memory config\n",
      "sysctl vm\n",
      "---------------------------------------------------------------------\n",
      "vm.admin_reserve_kbytes = 8192\n",
      "vm.block_dump = 0\n",
      "vm.compact_unevictable_allowed = 1\n",
      "vm.dirty_background_bytes = 0\n",
      "vm.dirty_background_ratio = 10\n",
      "vm.dirty_bytes = 0\n",
      "vm.dirty_expire_centisecs = 3000\n",
      "vm.dirty_ratio = 20\n",
      "vm.dirty_writeback_centisecs = 500\n",
      "vm.dirtytime_expire_seconds = 43200\n",
      "vm.extfrag_threshold = 500\n",
      "vm.hugetlb_shm_group = 0\n",
      "vm.laptop_mode = 0\n",
      "vm.legacy_va_layout = 0\n",
      "vm.lowmem_reserve_ratio = 256\t256\t32\t0\n",
      "vm.max_map_count = 65530\n",
      "vm.min_free_kbytes = 67584\n",
      "vm.mmap_min_addr = 65536\n",
      "vm.nr_hugepages = 0\n",
      "vm.nr_overcommit_hugepages = 0\n",
      "vm.oom_dump_tasks = 1\n",
      "vm.oom_kill_allocating_task = 0\n",
      "vm.overcommit_kbytes = 0\n",
      "vm.overcommit_memory = 0\n",
      "vm.overcommit_ratio = 50\n",
      "vm.page-cluster = 3\n",
      "vm.panic_on_oom = 0\n",
      "vm.percpu_pagelist_fraction = 0\n",
      "vm.stat_interval = 1\n",
      "vm.swappiness = 60\n",
      "vm.user_reserve_kbytes = 131072\n",
      "vm.vfs_cache_pressure = 100\n",
      "vm.watermark_scale_factor = 10\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "THP memory config\n",
      "cat /sys/kernel/mm/*transparent_hugepage/enabled\n",
      "---------------------------------------------------------------------\n",
      "[always] madvise never\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "cgroups\n",
      "cat /proc/self/cgroup\n",
      "---------------------------------------------------------------------\n",
      "14:name=systemd:/\n",
      "13:rdma:/\n",
      "12:pids:/\n",
      "11:hugetlb:/\n",
      "10:net_prio:/\n",
      "9:perf_event:/\n",
      "8:net_cls:/\n",
      "7:freezer:/\n",
      "6:devices:/\n",
      "5:memory:/\n",
      "4:blkio:/\n",
      "3:cpuacct:/\n",
      "2:cpu:/\n",
      "1:cpuset:/\n",
      "0::/\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "cgroup mem stats\n",
      "cat /sys/fs/cgroup/memory//memory.stat\n",
      "---------------------------------------------------------------------\n",
      "cache 1076391936\n",
      "rss 1936523264\n",
      "rss_huge 520093696\n",
      "shmem 381976576\n",
      "mapped_file 214106112\n",
      "dirty 0\n",
      "writeback 0\n",
      "swap 0\n",
      "pgpgin 114201879\n",
      "pgpgout 113597387\n",
      "pgfault 157022910\n",
      "pgmajfault 12144\n",
      "inactive_anon 265543680\n",
      "active_anon 2053378048\n",
      "inactive_file 392462336\n",
      "active_file 302350336\n",
      "unevictable 0\n",
      "hierarchical_memory_limit 9223372036854771712\n",
      "hierarchical_memsw_limit 9223372036854771712\n",
      "total_cache 1195065344\n",
      "total_rss 2098368512\n",
      "total_rss_huge 637534208\n",
      "total_shmem 384700416\n",
      "total_mapped_file 267632640\n",
      "total_dirty 0\n",
      "total_writeback 0\n",
      "total_swap 0\n",
      "total_pgpgin 114502212\n",
      "total_pgpgout 113856825\n",
      "total_pgfault 157431516\n",
      "total_pgmajfault 12375\n",
      "total_inactive_anon 268468224\n",
      "total_active_anon 2218635264\n",
      "total_inactive_file 413007872\n",
      "total_active_file 403447808\n",
      "total_unevictable 0\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "memory soft limit\n",
      "cat /sys/fs/cgroup/memory//memory.*soft_limit_in_bytes\n",
      "---------------------------------------------------------------------\n",
      "9223372036854771712\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "memory hard limit\n",
      "cat /sys/fs/cgroup/memory//memory.limit_in_bytes\n",
      "---------------------------------------------------------------------\n",
      "9223372036854771712\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "memory swap limit\n",
      "cat /sys/fs/cgroup/memory//memory.memsw.limit_in_bytes\n",
      "---------------------------------------------------------------------\n",
      "9223372036854771712\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "Container\n",
      "[ -e /.dockerenv ] || [ -e /.dockerinit ] || [ ! -z \"$container\" ] || grep -m 1 -E 'docker|lxc' /proc/1/cgroup > /dev/null && echo 'Detected'\n",
      "---------------------------------------------------------------------\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "init process\n",
      "head -n 1 /proc/1/sched | cut -d ' ' -f 1\n",
      "---------------------------------------------------------------------\n",
      "init\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "SGE Submit\n",
      "which qsub\n",
      "---------------------------------------------------------------------\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "LSF Submit\n",
      "which bsub\n",
      "---------------------------------------------------------------------\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "BCL2FASTQ 1\n",
      "which configureBclToFastq.pl\n",
      "---------------------------------------------------------------------\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "BCL2FASTQ 2\n",
      "which bcl2fastq\n",
      "---------------------------------------------------------------------\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "Java\n",
      "which java\n",
      "---------------------------------------------------------------------\n",
      "/usr/bin/java\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "Java Version\n",
      "java -version 2>&1 | cat\n",
      "---------------------------------------------------------------------\n",
      "openjdk version \"14.0.2\" 2020-07-14\n",
      "OpenJDK Runtime Environment (build 14.0.2+12-Ubuntu-120.04)\n",
      "OpenJDK 64-Bit Server VM (build 14.0.2+12-Ubuntu-120.04, mixed mode, sharing)\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "10X Refdata\n",
      "echo $TENX_REFDATA\n",
      "---------------------------------------------------------------------\n",
      "\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "10X Refdata Version\n",
      "cat $TENX_REFDATA/version\n",
      "---------------------------------------------------------------------\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "qconf\n",
      "which qconf\n",
      "---------------------------------------------------------------------\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "slurm info\n",
      "sinfo -O nodes,maxcpuspernode,memory,time\n",
      "---------------------------------------------------------------------\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "MRP\n",
      "mrp --version\n",
      "---------------------------------------------------------------------\n",
      "v4.0.6\n",
      "=====================================================================\n",
      "\n",
      "=====================================================================\n",
      "mrp templates\n",
      "ls $(dirname $(dirname $(which mrp)))/jobmanagers/*.template\n",
      "---------------------------------------------------------------------\n",
      "=====================================================================\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! tools/cellranger-6.1.2/bin/cellranger sitecheck"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now download and split the SRA files. This will result in 3 files per SRA-run (SRR ID), the read1, read2 and the index file:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir reads\n",
    "! mkdir reads/SUM159DMSO\n",
    "! mkdir reads/SUM159RDMSO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cd reads/SUM159DMSO/ && fastq-dump --split-files --gzip SRR9050597\n",
    "! cd reads/SUM159DMSO/ && fastq-dump --split-files --gzip SRR9050598\n",
    "! cd reads/SUM159DMSO/ && fastq-dump --split-files --gzip SRR9050599\n",
    "! cd reads/SUM159DMSO/ && fastq-dump --split-files --gzip SRR9050600"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cd reads/SUM159RDMSO/ && fastq-dump --split-files --gzip SRR9050605\n",
    "! cd reads/SUM159RDMSO/ && fastq-dump --split-files --gzip SRR9050606\n",
    "! cd reads/SUM159RDMSO/ && fastq-dump --split-files --gzip SRR9050607\n",
    "! cd reads/SUM159RDMSO/ && fastq-dump --split-files --gzip SRR9050608"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Unfortunately the SRA-format is incompatible with cellranger. You need to manually process it according to https://kb.10xgenomics.com/hc/en-us/articles/115003802691-How-do-I-prepare-Sequence-Read-Archive-SRA-data-from-NCBI-for-Cell-Ranger- .\n",
    "\n",
    "Based on the read length and the attributes listed in the meta data, we know that:\n",
    "\n",
    "    SRR9291388_1.fastq is Read 1\n",
    "    SRR9291388_2.fastq is Read 2\n",
    "    SRR9291388_3.fastq is Index 1\n",
    "\n",
    "Cell Ranger requires FASTQ file names to follow the bcl2fastq file naming convention.\n",
    "\n",
    "[Sample Name]_S1_L00[Lane Number]_[Read Type]_001.fastq.gz\n",
    "\n",
    "Where Read Type is one of:\n",
    "\n",
    "    I1: Sample index read (optional)\n",
    "    I2: Sample index read (optional)\n",
    "    R1: Read 1 (required)\n",
    "    R2: Read 2 (required)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mv sc/SUM159DMSO/SRR9050597_1.fastq.gz sc/SUM159DMSO/SRR9050597_S1_L001_R1_001.fastq.gz\n",
    "! mv sc/SUM159DMSO/SRR9050597_2.fastq.gz sc/SUM159DMSO/SRR9050597_S1_L001_R2_001.fastq.gz\n",
    "\n",
    "! mv sc/SUM159DMSO/SRR9050598_1.fastq.gz sc/SUM159DMSO/SRR9050598_S1_L001_R1_001.fastq.gz\n",
    "! mv sc/SUM159DMSO/SRR9050598_2.fastq.gz sc/SUM159DMSO/SRR9050598_S1_L001_R2_001.fastq.gz\n",
    "\n",
    "! mv sc/SUM159DMSO/SRR9050599_1.fastq.gz sc/SUM159DMSO/SRR9050599_S1_L001_R1_001.fastq.gz\n",
    "! mv sc/SUM159DMSO/SRR9050599_2.fastq.gz sc/SUM159DMSO/SRR9050599_S1_L001_R2_001.fastq.gz\n",
    "\n",
    "! mv sc/SUM159DMSO/SRR9050600_1.fastq.gz sc/SUM159DMSO/SRR9050600_S1_L001_R1_001.fastq.gz\n",
    "! mv sc/SUM159DMSO/SRR9050600_2.fastq.gz sc/SUM159DMSO/SRR9050600_S1_L001_R2_001.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mv sc/SUM159RDMSO/SRR9050605_1.fastq.gz sc/SUM159RDMSO/SRR9050605_S1_L001_R1_001.fastq.gz\n",
    "! mv sc/SUM159RDMSO/SRR9050605_2.fastq.gz sc/SUM159RDMSO/SRR9050605_S1_L001_R2_001.fastq.gz\n",
    "\n",
    "! mv sc/SUM159RDMSO/SRR9050606_1.fastq.gz sc/SUM159RDMSO/SRR9050606_S1_L001_R1_001.fastq.gz\n",
    "! mv sc/SUM159RDMSO/SRR9050606_2.fastq.gz sc/SUM159RDMSO/SRR9050606_S1_L001_R2_001.fastq.gz\n",
    "\n",
    "! mv sc/SUM159RDMSO/SRR9050607_1.fastq.gz sc/SUM159RDMSO/SRR9050607_S1_L001_R1_001.fastq.gz\n",
    "! mv sc/SUM159RDMSO/SRR9050607_2.fastq.gz sc/SUM159RDMSO/SRR9050607_S1_L001_R2_001.fastq.gz\n",
    "\n",
    "! mv sc/SUM159RDMSO/SRR9050608_1.fastq.gz sc/SUM159RDMSO/SRR9050608_S1_L001_R1_001.fastq.gz\n",
    "! mv sc/SUM159RDMSO/SRR9050608_2.fastq.gz sc/SUM159RDMSO/SRR9050608_S1_L001_R2_001.fastq.gz\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Consult the presentation for structuring the sc folder and the calls file. When properly set up, it should look like this: (the folders in the calls subfolder in sc will only appear after having run cellranger!)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"../images/folder_layout.png\" />"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You can now run cellranger to process the individual samples:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! bash sc/calls/calls.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After having executed cellranger, we must merge both datasets:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Martian Runtime - v4.0.6\n",
      "Serving UI at http://spectre3:33385?auth=n6xPWtniVBFNIQwVddnvZKtu8SkPqAVUPPwjmd2wns4\n",
      "\n",
      "Running preflight checks (please wait)...\n",
      "2022-01-12 21:18:28 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.PARSE_AGGR_CSV\n",
      "2022-01-12 21:18:28 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.PARSE_AGGR_CSV.fork0.chnk0.main\n",
      "2022-01-12 21:18:28 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.PARSE_AGGR_CSV\n",
      "2022-01-12 21:18:29 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CHECK_MOLECULE_INFO_VERSION\n",
      "2022-01-12 21:18:29 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CHECK_MOLECULE_INFO_VERSION.fork0.split\n",
      "2022-01-12 21:18:42 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CHECK_MOLECULE_INFO_VERSION\n",
      "2022-01-12 21:18:42 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CHECK_MOLECULE_INFO_VERSION.fork0.chnk0.main\n",
      "2022-01-12 21:18:42 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CHECK_MOLECULE_INFO_VERSION.fork0.chnk1.main\n",
      "2022-01-12 21:18:52 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CHECK_MOLECULE_INFO_VERSION\n",
      "2022-01-12 21:18:52 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CHECK_MOLECULE_INFO_VERSION.fork0.join\n",
      "2022-01-12 21:18:58 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.VDJ_AGGR.MATCH_VDJ_AGGR_OUTS\n",
      "2022-01-12 21:18:58 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.VDJ_AGGR.COPY_VDJ_REFERENCE\n",
      "2022-01-12 21:19:06 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CHECK_MOLECULE_INFO_VERSION\n",
      "2022-01-12 21:19:15 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SETUP_SAMPLES\n",
      "2022-01-12 21:19:15 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SETUP_SAMPLES.fork0.chnk0.main\n",
      "2022-01-12 21:19:22 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SETUP_SAMPLES\n",
      "2022-01-12 21:19:23 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.MERGE_MOLECULES\n",
      "2022-01-12 21:19:23 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.MERGE_MOLECULES.fork0.split\n",
      "2022-01-12 21:19:37 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.MERGE_MOLECULES\n",
      "2022-01-12 21:19:37 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.MERGE_MOLECULES.fork0.chnk0.main\n",
      "2022-01-12 21:19:37 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.MERGE_MOLECULES.fork0.chnk1.main\n",
      "2022-01-12 21:19:52 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.MERGE_MOLECULES\n",
      "2022-01-12 21:19:52 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.MERGE_MOLECULES.fork0.join\n",
      "2022-01-12 21:20:34 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.MERGE_MOLECULES\n",
      "2022-01-12 21:20:34 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.NORMALIZE_DEPTH\n",
      "2022-01-12 21:20:34 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.NORMALIZE_DEPTH.fork0.split\n",
      "2022-01-12 21:20:43 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.NORMALIZE_DEPTH\n",
      "2022-01-12 21:20:43 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.NORMALIZE_DEPTH.fork0.chnk0.main\n",
      "2022-01-12 21:21:04 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.NORMALIZE_DEPTH\n",
      "2022-01-12 21:21:04 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.NORMALIZE_DEPTH.fork0.join\n",
      "2022-01-12 21:21:13 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.NORMALIZE_DEPTH\n",
      "2022-01-12 21:21:13 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.WRITE_MATRICES\n",
      "2022-01-12 21:21:13 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.WRITE_MATRICES.fork0.split\n",
      "2022-01-12 21:21:21 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.WRITE_MATRICES\n",
      "2022-01-12 21:21:21 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.WRITE_MATRICES.fork0.join\n",
      "2022-01-12 21:21:39 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.WRITE_MATRICES\n",
      "2022-01-12 21:21:39 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.ANALYZER_PREFLIGHT\n",
      "2022-01-12 21:21:39 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.ANALYZER_PREFLIGHT.fork0.chnk0.main\n",
      "2022-01-12 21:21:48 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.ANALYZER_PREFLIGHT\n",
      "2022-01-12 21:21:48 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.PREPROCESS_MATRIX\n",
      "2022-01-12 21:21:48 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.PREPROCESS_MATRIX.fork0.split\n",
      "2022-01-12 21:22:02 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.PREPROCESS_MATRIX\n",
      "2022-01-12 21:22:02 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.PREPROCESS_MATRIX.fork0.join\n",
      "2022-01-12 21:22:19 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.PREPROCESS_MATRIX\n",
      "2022-01-12 21:22:19 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_PCA\n",
      "2022-01-12 21:22:19 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_PCA.fork0.split\n",
      "2022-01-12 21:22:19 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_FBPCA\n",
      "2022-01-12 21:22:19 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_MULTIGENOME_ANALYSIS\n",
      "2022-01-12 21:22:20 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_MULTIGENOME_ANALYSIS.fork0.split\n",
      "2022-01-12 21:22:20 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.CORRECT_CHEMISTRY_BATCH\n",
      "2022-01-12 21:22:20 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_PCA\n",
      "2022-01-12 21:22:20 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_PCA.fork0.join\n",
      "2022-01-12 21:22:27 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_PCA\n",
      "2022-01-12 21:22:27 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.CHOOSE_DIMENSION_REDUCTION_OUTPUT\n",
      "2022-01-12 21:22:27 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.CHOOSE_DIMENSION_REDUCTION_OUTPUT.fork0.chnk0.main\n",
      "2022-01-12 21:22:28 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.CHOOSE_DIMENSION_REDUCTION_OUTPUT\n",
      "2022-01-12 21:22:28 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS\n",
      "2022-01-12 21:22:28 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.split\n",
      "2022-01-12 21:22:28 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_TSNE\n",
      "2022-01-12 21:22:28 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_TSNE.fork0.split\n",
      "2022-01-12 21:22:28 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_GRAPH_CLUSTERING\n",
      "2022-01-12 21:22:28 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_GRAPH_CLUSTERING.fork0.split\n",
      "2022-01-12 21:22:28 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_UMAP\n",
      "2022-01-12 21:22:28 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_UMAP.fork0.split\n",
      "2022-01-12 21:22:29 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_TSNE\n",
      "2022-01-12 21:22:29 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_TSNE.fork0.chnk0.main\n",
      "2022-01-12 21:22:29 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_GRAPH_CLUSTERING\n",
      "2022-01-12 21:22:29 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_GRAPH_CLUSTERING.fork0.join\n",
      "2022-01-12 21:22:29 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_UMAP\n",
      "2022-01-12 21:22:29 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_UMAP.fork0.chnk0.main\n",
      "2022-01-12 21:22:30 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_MULTIGENOME_ANALYSIS\n",
      "2022-01-12 21:22:30 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_MULTIGENOME_ANALYSIS.fork0.join\n",
      "2022-01-12 21:22:33 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_TSNE\n",
      "2022-01-12 21:22:33 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_TSNE.fork0.join\n",
      "2022-01-12 21:22:33 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_TSNE\n",
      "2022-01-12 21:22:33 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_GRAPH_CLUSTERING\n",
      "2022-01-12 21:22:38 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS\n",
      "2022-01-12 21:22:38 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.chnk0.main\n",
      "2022-01-12 21:22:38 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.chnk1.main\n",
      "2022-01-12 21:22:38 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.chnk2.main\n",
      "2022-01-12 21:22:38 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.chnk3.main\n",
      "2022-01-12 21:22:38 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.chnk4.main\n",
      "2022-01-12 21:22:38 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.chnk5.main\n",
      "2022-01-12 21:22:38 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.chnk6.main\n",
      "2022-01-12 21:22:38 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.chnk7.main\n",
      "2022-01-12 21:22:38 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.chnk8.main\n",
      "2022-01-12 21:22:39 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_MULTIGENOME_ANALYSIS\n",
      "2022-01-12 21:22:44 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_UMAP\n",
      "2022-01-12 21:22:44 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_UMAP.fork0.join\n",
      "2022-01-12 21:22:51 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_UMAP\n",
      "2022-01-12 21:23:00 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS\n",
      "2022-01-12 21:23:00 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS.fork0.join\n",
      "2022-01-12 21:23:11 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_KMEANS\n",
      "2022-01-12 21:23:12 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.COMBINE_CLUSTERING\n",
      "2022-01-12 21:23:12 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.COMBINE_CLUSTERING.fork0.chnk0.main\n",
      "2022-01-12 21:23:17 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.COMBINE_CLUSTERING\n",
      "2022-01-12 21:23:18 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION\n",
      "2022-01-12 21:23:18 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION.fork0.split\n",
      "2022-01-12 21:23:18 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION\n",
      "2022-01-12 21:23:18 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION.fork0.chnk0.main\n",
      "2022-01-12 21:23:18 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION.fork0.chnk1.main\n",
      "2022-01-12 21:23:18 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION.fork0.chnk2.main\n",
      "2022-01-12 21:23:18 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION.fork0.chnk3.main\n",
      "2022-01-12 21:23:18 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION.fork0.chnk4.main\n",
      "2022-01-12 21:23:18 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION.fork0.chnk5.main\n",
      "2022-01-12 21:23:18 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION.fork0.chnk6.main\n",
      "2022-01-12 21:23:19 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION.fork0.chnk7.main\n",
      "2022-01-12 21:23:51 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION\n",
      "2022-01-12 21:23:51 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION.fork0.join\n",
      "2022-01-12 21:23:54 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.RUN_DIFFERENTIAL_EXPRESSION\n",
      "2022-01-12 21:23:54 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.SUMMARIZE_ANALYSIS\n",
      "2022-01-12 21:23:54 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.SUMMARIZE_ANALYSIS.fork0.split\n",
      "2022-01-12 21:23:59 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.SUMMARIZE_ANALYSIS\n",
      "2022-01-12 21:23:59 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.SUMMARIZE_ANALYSIS.fork0.chnk0.main\n",
      "2022-01-12 21:24:09 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.SUMMARIZE_ANALYSIS\n",
      "2022-01-12 21:24:09 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.SUMMARIZE_ANALYSIS.fork0.join\n",
      "2022-01-12 21:24:16 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SC_RNA_ANALYZER.SUMMARIZE_ANALYSIS\n",
      "2022-01-12 21:24:18 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SUMMARIZE_AGGREGATED_REPORTS\n",
      "2022-01-12 21:24:18 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SUMMARIZE_AGGREGATED_REPORTS.fork0.split\n",
      "2022-01-12 21:24:40 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SUMMARIZE_AGGREGATED_REPORTS\n",
      "2022-01-12 21:24:40 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SUMMARIZE_AGGREGATED_REPORTS.fork0.join\n",
      "2022-01-12 21:25:08 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.SC_RNA_AGGREGATOR.SUMMARIZE_AGGREGATED_REPORTS\n",
      "2022-01-12 21:25:09 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.BUILD_COMBINED_WEB_SUMMARY\n",
      "2022-01-12 21:25:09 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.BUILD_COMBINED_WEB_SUMMARY.fork0.chnk0.main\n",
      "2022-01-12 21:25:09 [runtime] (ready)           ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CLOUPE_PREPROCESS\n",
      "2022-01-12 21:25:09 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CLOUPE_PREPROCESS.fork0.split\n",
      "2022-01-12 21:25:11 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.BUILD_COMBINED_WEB_SUMMARY\n",
      "2022-01-12 21:25:18 [runtime] (split_complete)  ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CLOUPE_PREPROCESS\n",
      "2022-01-12 21:25:18 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CLOUPE_PREPROCESS.fork0.chnk0.main\n",
      "2022-01-12 21:25:47 [runtime] (chunks_complete) ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CLOUPE_PREPROCESS\n",
      "2022-01-12 21:25:47 [runtime] (run:local)       ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CLOUPE_PREPROCESS.fork0.join\n",
      "2022-01-12 21:25:57 [runtime] (join_complete)   ID.SUM159DMSO_RDMSO.SC_RNA_AGGREGATOR_CS.COUNT_AGGR.CLOUPE_PREPROCESS\n",
      "\n",
      "Outputs:\n",
      "- Copy of the input aggregation CSV: /mnt/w/game_cibog/sc/aggr_calls/SUM159DMSO_RDMSO/outs/aggregation.csv\n",
      "- Aggregation metrics summary HTML:  /mnt/w/game_cibog/sc/aggr_calls/SUM159DMSO_RDMSO/outs/web_summary.html\n",
      "- count:\n",
      "    Aggregation metrics summary JSON:       /mnt/w/game_cibog/sc/aggr_calls/SUM159DMSO_RDMSO/outs/count/summary.json\n",
      "    Secondary analysis output CSV:          /mnt/w/game_cibog/sc/aggr_calls/SUM159DMSO_RDMSO/outs/count/analysis\n",
      "    Filtered feature-barcode matrices MEX:  /mnt/w/game_cibog/sc/aggr_calls/SUM159DMSO_RDMSO/outs/count/filtered_feature_bc_matrix\n",
      "    Filtered feature-barcode matrices HDF5: /mnt/w/game_cibog/sc/aggr_calls/SUM159DMSO_RDMSO/outs/count/filtered_feature_bc_matrix.h5\n",
      "    Loupe Browser file:                     /mnt/w/game_cibog/sc/aggr_calls/SUM159DMSO_RDMSO/outs/count/cloupe.cloupe\n",
      "- vdj_t:                             null\n",
      "- vdj_b:                             null\n",
      "- V(D)J reference:                   null\n",
      "\n",
      "Waiting 6 seconds for UI to do final refresh.\n",
      "Pipestance completed successfully!\n",
      "\n",
      "2022-01-12 21:26:05 Shutting down.\n"
     ]
    }
   ],
   "source": [
    "! cd sc/aggr_calls/ && ../../tools/cellranger-6.1.2/bin/cellranger aggr --localmem 30 --localcores 8 --csv aggr_samples.csv --id SUM159DMSO_RDMSO"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "31f2aee4e71d21fbe5cf8b01ff0e069b9275f58929596ceb00d14d90e3e16cd6"
  },
  "kernelspec": {
   "display_name": "Python 3.8.10 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
